cmd/kyma/test/run/cmd.go

```diff
@@ -42,17 +50,25 @@ To execute all test defintions, run ` + "`kyma test run -n example-test`" + `.
 	}
 
 	cobraCmd.Flags().StringVarP(&o.Name, "name", "n", "", "Specifies the name of the new test suite. If you don't specify the value for the `-n` flag, the name of the test suite will be autogenerated.")
-	cobraCmd.Flags().Int64VarP(&o.ExecutionCount, "count", "c", 1, "Defines how many times every test should be executed. \"count\" and \"max-retries\" flags are mutually exclusive.")
-	cobraCmd.Flags().Int64VarP(&o.MaxRetries, "max-retries", "", 0, "Defines how many times a given test is retried when it fails. A suite is marked with a \"succeeded\" status even if some tests failed at first and then finally succeeded. The default value of 0 means that there are no retries of a given test.")
+	cobraCmd.Flags().Int64VarP(&o.ExecutionCount, "count", "c", 1, `Defines how many times every test should be executed. "count" and "max-retries" flags are mutually exclusive.`)
+	cobraCmd.Flags().Int64VarP(&o.MaxRetries, "max-retries", "", 0, `Defines how many times a given test is retried when it fails. A suite is marked with a "succeeded" status even if some tests failed at first and then finally succeeded. The default value of 0 means that there are no retries of a given test.`)
 	cobraCmd.Flags().Int64VarP(&o.Concurrency, "concurrency", "", 1, "Specifies the number of tests to be executed in parallel.")
+	cobraCmd.Flags().DurationVar(&o.Timeout, "timeout", 10*time.Minute, `Maximum time during which the test suite is being watched. Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".`)
+	cobraCmd.Flags().BoolVarP(&o.Watch, "watch", "w", o.Watch, "Watch the status of the test suite until it finishes or the defined `--timeout` occurs.")
 	cobraCmd.Flags().Bool("help", false, "Displays help for the command.")
 	return cobraCmd
 }
 
 func (cmd *command) Run(args []string) error {
 	var err error
-	if cmd.K8s, err = kube.NewFromConfig("", cmd.KubeconfigPath); err != nil {
-		return errors.Wrap(err, "Could not initialize the Kubernetes client. Make sure your kubeconfig is valid.")
+	if cmd.opts.Watch {
+		if cmd.K8s, err = kube.NewFromConfigWithTimeout("", cmd.KubeconfigPath, cmd.opts.Timeout); err != nil {
```

Comment:
The apiserver commands should have a timeout always I guess. And that timeout should not be the one used for the watch

Reply:
dicussed